inspirehep/records/api/base.py

```diff
@@ -264,8 +264,16 @@ def get_records_by_pids(cls, pids):
         for data in query.yield_per(100):
             yield cls(data.json)
 
+    @classmethod
+    def get_class_from_schema(cls, data):
+        type_from_schema = PidStoreBase.get_pid_type_from_schema(data["$schema"])
+        return cls.get_subclasses().get(type_from_schema, cls)
+
     @classmethod
     def create(cls, data, **kwargs):
+        subclass = cls.get_class_from_schema(data)
+        if subclass != cls:
```

Comment:
You should also add tests to check if it will behave correctly when you call Create from correct subclass :)